mne/io/boxy/boxy.py

```diff
@@ -184,22 +221,46 @@ def __init__(self, fname, preload=False, verbose=None):
         ###will label them based on ac, dc, and ph data###
         boxy_coords = []
         boxy_labels = []
-        data_types = ['AC','DC','Ph']
-        total_chans = detect_num*source_num
-        for i_type in data_types:
-            for i_coord in range(len(source_coords[0:total_chans])):
-                boxy_coords.append(np.mean(
-                    np.vstack((source_coords[i_coord], detect_coords[i_coord])),
-                    axis=0).tolist() + source_coords[i_coord] + 
-                    detect_coords[i_coord] + [chan_wavelength[i_coord]] + [0] + [0])
-                boxy_labels.append('S' + 
-                                       str(unique_source_labels.index(source_label[i_coord])+1)
-                                       + '_D' + 
-                                       str(unique_detect_labels.index(detect_label[i_coord])+1) 
-                                       + ' ' + chan_wavelength[i_coord] + ' ' + i_type)
+        mrk_coords = []
+        mrk_labels = []
+        mtg_start = []
+        mtg_end = []
+        mtg_src_num = []
+        mtg_det_num = []
+        blk_num = [len(blk) for blk in blk_names]
+        for mtg_num, i_mtg in enumerate(mtg_chan_num,0):
+            start = int(np.sum(mtg_chan_num[:mtg_num]))
+            end = int(np.sum(mtg_chan_num[:mtg_num+1]))
+            ###we will also organise some data for each montage###
+            start_blk = int(np.sum(blk_num[:mtg_num]))
+            ###get stop and stop lines for each montage###
+            mtg_start.append(start_line[start_blk])
+            mtg_end.append(end_line[start_blk])
+            ###get source and detector numbers for each montage###
+            mtg_src_num.append(source_num[start_blk])
+            mtg_det_num.append(detect_num[start_blk])
+            for i_blk in blk_names[mtg_num]:
+                for i_type in data_types:
+                    for i_coord in range(start,end):
+                        boxy_coords.append(np.mean(
+                            np.vstack((source_coords[i_coord], detect_coords[i_coord])),
+                            axis=0).tolist() + source_coords[i_coord] + 
+                            detect_coords[i_coord] + [chan_wavelength[i_coord]] + [0] + [0])
+                        boxy_labels.append('S' + str(unique_source_labels[mtg_num].index(
+                                            source_label[i_coord])+1) + '_D' + 
+                                            str(unique_detect_labels[mtg_num].index(
+                                            detect_label[i_coord])+1) + ' ' + 
+                                            chan_wavelength[i_coord] + ' ' + 
+                                            mtg_names[mtg_num] + i_blk[1:])
```

Comment:
lets remove the mtg names and block number so our channel names match the fNIRS standard S1 D1 wvl

Reply:
okay but how do we want to keep track of which channels belong to which block and montage? We'll also end up with duplicate channel names and mne will attach a number to each one anyway so that the names are different.

Reply:
channels should stay the same over blocks, channels should be different in different montages, they should be in different locations on the head, but now should also get different numbers in the D1 S1 format, so after first data from montage A is loaded, start with the subsequent D and S numbers so each channel gets unique name,

i noticed it appends a number, one of the reasons we want to only load one Data type (ac, dc, ph) at a time

Reply:
ah okay. yeah I was treating the first source for each montage as S1, rather than referring to the master list in the elp file. I'll change that and then close this once done.

I'll resolve the above comment (regarding AC, DC, and ph) once I address this as well.